### PR TITLE
Replace casts/assignments with hard-coded memcpy operations

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: hpx/traits/serialization_access_data.hpp [hunk 1]
```diff
@@ -96,19 +96,20 @@ namespace hpx { namespace traits
...
             case 4:
-                *static_cast<std::uint32_t*>(dest) =
-                    *static_cast<std::uint32_t const*>(address);
+                std::memcpy(dest, address, 4);
                 break;
 
             case 2:
-                *static_cast<std::uint16_t*>(dest) =
-                    *static_cast<std::uint16_t const*>(address);
+                std::memcpy(dest, address, 2);
```
Comment (taeguk) [inline hpx/traits/serialization_access_data.hpp]: Anyway, ```std::memcpy``` is performed in ```default:```. So, as I think, there is no need to be handled separately.

Comment (hkaiser) [reply hpx/traits/serialization_access_data.hpp]: The point of repeating the `memcpy()` invocations with constant byte sizes is that compilers usually optimize those, whereas the generic `memcpy(d, s, N)` can't be optimized as aggressively.

Comment (taeguk) [reply hpx/traits/serialization_access_data.hpp]: @hkaiser Oh, makes sense! I got it 👍 You're right.

Comment (taeguk) [reply hpx/traits/serialization_access_data.hpp]: Does the benefit of compiler optimizations exceed the overhead of switch-case statement?

Comment (hkaiser) [reply hpx/traits/serialization_access_data.hpp]: The measurements show that serialization is up to 10% faster with this change.

**Pull request merged**